### PR TITLE
linux-libre_latest: update automatically; fix build

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

Alternative to #74726.

This should mean that Linux-libre is updated whenever @NeQuissimus updates the kernel, AND whenever update scripts are run — I think r-ryantm does this?

Between these two scenarios, I’m hoping this should be run often enough that the libre kernels almost always build, even when new kernel series are added in Nixpkgs before Linux-libre supports them.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by worldofpeace** [comment]
> AND whenever update scripts are run — I think r-ryantm does this?

I believe this is handled completely differently. These updates scripts are usually only ran by humans.

**Comment by alyssais** [comment]
> > AND whenever update scripts are run — I think r-ryantm does this?
> 
> I believe this is handled completely differently. These updates scripts are usually only ran by humans.

Oh, you’re right. :(

**Comment by worldofpeace** [comment]
It seems good to me @alyssais, but I'm not sure about adding to `pkgs/os-specific/linux/kernel/update.sh`. Maintainers of kernel have to be ok with picking up this one, as it makes it update the libre one.

**Comment by worldofpeace** [comment]
Also :heart: for the kernel in NixOS being integrated so expectantly @NeQuissimus.

**Comment by alyssais** [comment]
> Maintainers of kernel have to be ok with picking up this one, as it makes it update the libre one.

Absolutely, hence the request for a review from @NeQuissimus. fwiw, I don’t think the kernel maintainers have any responsibility for ensuring Linux-libre build after updating it — this is just something that needs to be done at the same time. The alternative would be I write a bot to do it, which would just be more annoying for everyone I suspect.

**Comment by NeQuissimus** [comment]
I'll put it this way:
I can run the script, for sure. This will update the kernel. But the build itself I cannot run, it is far more time-intensive than the vanilla kernels.
So if the libre kernel breaks from my commits, I will not be able to fix it.
If everybody is OK with that, let's do this.

**Review by NeQuissimus — APPROVED** [review]

**Comment by alyssais** [comment]
That sounds good to me.

**Comment by alyssais** [comment]
Pushed a tiny ﬁxup I noticed just after merging: d770ae120293e8e796ab7c974c123f143bc9d5a1

**Comment by alyssais** [comment]
Backported the linux-libre scripts update in c337a7423bc.

**Pull request merged**
